Foundation/ProcessInfo.swift

```diff
@@ -104,12 +104,24 @@ open class ProcessInfo: NSObject {
         }
         versionString = productVersion._swiftObject
 #elseif os(Windows)
-        var siVersionInfo: OSVERSIONINFOW = OSVERSIONINFOW()
-        siVersionInfo.dwOSVersionInfoSize = DWORD(MemoryLayout<OSVERSIONINFOEXW>.size)
-        if GetVersionExW(&siVersionInfo) == FALSE {
-          return OperatingSystemVersion(majorVersion: fallbackMajor, minorVersion: fallbackMinor, patchVersion: fallbackPatch)
+        guard let ntdll = "ntdll.dll".withCString(encodedAs: UTF16.self, { LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32)) }) else {
```

Comment:
Make the closure trailing, the `guard let` and add a newline after the brace perhaps?

```suggestion
        guard let ntdll = "ntdll.dll".withCString(encodedAs: UTF16.self) {
          LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
        } else {
```

Reply:
Thank you for your advice again. Your suggestion's form requires additonal parentheses before `"ntdll.dll"` and `else` like this:
```swift
        guard let ntdll = ("ntdll.dll".withCString(encodedAs: UTF16.self) {
          LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
        }) else {
            return OperatingSystemVersion(majorVersion: fallbackMajor, minorVersion: fallbackMinor, patchVersion: fallbackPatch)
        }
```
Without those, the compiler gets confused.
```
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:107:74: error: expected 'else' after 'guard' condition
        guard let ntdll = "ntdll.dll".withCString(encodedAs: UTF16.self) {
                                                                         ^
                                                                         else
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:108:26: error: anonymous closure argument not contained in a closure
          LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
                         ^
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:109:10: error: consecutive statements on a line must be separated by ';'
        } else {
         ^
         ;
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:109:11: error: expected expression
        } else {
          ^
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:107:39: error: cannot invoke 'withCString' with an argument list of type '(encodedAs: UTF16.Type)'
        guard let ntdll = "ntdll.dll".withCString(encodedAs: UTF16.self) {
                                      ^
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:109:16: error: closure expression is unused
        } else {
               ^
C:/swift/swift-corelibs-foundation/Foundation/ProcessInfo.swift:109:16: note: did you mean to use a 'do' statement?
        } else {
               ^
               do
```
Do you prefer your suggestion with additonal parentheses or a form in PR?

Reply:
The wrapped form is definitely easier to read IMO.